assets/src/dashboard/animations/effects/pan/index.js

```diff
@@ -0,0 +1,127 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { getBox, dataToEditorY } from '../../../../edit-story/units/dimensions';
+import SimpleAnimation from '../../parts/simpleAnimation';
+import { DIRECTION, ANIMATION_EFFECTS } from '../../constants';
+import { PAGE_HEIGHT, PAGE_WIDTH, FULLBLEED_RATIO } from '../../../constants';
+
+const FULLBLEED_HEIGHT = PAGE_WIDTH / FULLBLEED_RATIO;
+const DANGER_ZONE_HEIGHT = (FULLBLEED_HEIGHT - PAGE_HEIGHT) / 2;
+
+function getTargetScale({ width, height }) {
+  if (width < PAGE_WIDTH || height < FULLBLEED_HEIGHT) {
+    const scaleFactor = 1.25;
+
+    const widthScale = width < PAGE_WIDTH ? PAGE_WIDTH / width : 1;
+    const heightScale =
+      height < FULLBLEED_HEIGHT ? FULLBLEED_HEIGHT / height : 1;
+
+    return Math.max(widthScale, heightScale) * scaleFactor;
+  }
+
+  return 1;
+}
+
+export function EffectPan({
+  panDir = DIRECTION.RIGHT_TO_LEFT,
+  duration = 500,
+  delay,
+  easing,
+  element,
+}) {
+  const timings = {
+    fill: 'both',
+    duration,
+    delay,
+    easing,
+  };
+
+  const scale = getTargetScale(element);
+  const scaledElement = {
+    x: element.x,
+    y: element.y,
+    width: element.width * scale,
+    height: element.height * scale,
+  };
+
+  const { x, y, width, height } = getBox(scaledElement, 100, 100);
+  const dangerZoneOffset = dataToEditorY(DANGER_ZONE_HEIGHT, 100);
+
+  const alignTop = -((dangerZoneOffset + y) / height) * 100.0;
+  const alignRight = ((100 - (x + width)) / width) * 100.0;
+  const alignLeft = -(x / width) * 100.0;
+  const alignBottom =
+    ((100 - (y + height) + dangerZoneOffset) / height) * 100.0;
+
+  let startX, startY, endX, endY;
+
+  switch (panDir) {
+    case DIRECTION.TOP_TO_BOTTOM:
+      startX = (alignLeft + alignRight) / 2;
+      startY = alignTop;
+      endX = (alignLeft + alignRight) / 2;
+      endY = alignBottom;
+      break;
+    case DIRECTION.BOTTOM_TO_TOP:
+      startX = (alignLeft + alignRight) / 2;
+      startY = alignBottom;
+      endX = (alignLeft + alignRight) / 2;
+      endY = alignTop;
+      break;
+    case DIRECTION.LEFT_TO_RIGHT:
+      startX = alignLeft;
+      startY = alignTop;
+      endX = alignRight;
+      endY = alignTop;
+      break;
+    default:
+      // RIGHT_TO_LEFT
+      startX = alignRight;
+      startY = alignTop;
+      endX = alignLeft;
+      endY = alignTop;
+  }
+
+  const animationName = `direction-${panDir}-scale-${scale}-${ANIMATION_EFFECTS.PAN}`;
+  const keyframes = {
+    transform: [
+      `translate(${startX}%, ${startY}%) scale(${scale})`,
+      `translate(${endX}%, ${endY}%) scale(${scale})`,
+    ],
+    'transform-origin': 'left top',
```

Comment:
AMP doesn't like `transformOrigin` so I was forced to use `transform-origin` which causes react to ask "Did you mean `transformOrigin`?" 😱 But it's a small bug on the AMP side because looking at their code it looks like they meant to include `transformOrigin` in the whitelist of animation properties but forgot... so I'll be submitting a small patch to AMP to see if I can get it fixed.